app/src/main/java/com/belatrixsf/allstars/ui/stars/StarsListFragment.java

```diff
@@ -81,13 +84,37 @@ protected void initDependencies(AllStarsApplication allStarsApplication) {
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         initViews();
-        if (getArguments() != null && getArguments().containsKey(StarsListActivity.USER_ID) && getArguments().containsKey(StarsListActivity.SUBCATEGORY_ID)) {
+        boolean hasArguments = (getArguments() != null && getArguments().containsKey(StarsListActivity.USER_ID) && getArguments().containsKey(StarsListActivity.SUBCATEGORY_ID));
+        if (savedInstanceState != null && hasArguments) {
+            restoreState(savedInstanceState);
+            Integer userId = getArguments().getInt(StarsListActivity.USER_ID);
+            Integer categoryId = getArguments().getInt(StarsListActivity.SUBCATEGORY_ID);
+            starsListPresenter.getStars(userId, categoryId);
+        }else if (hasArguments) {
             Integer userId = getArguments().getInt(StarsListActivity.USER_ID);
             Integer categoryId = getArguments().getInt(StarsListActivity.SUBCATEGORY_ID);
             starsListPresenter.getStars(userId, categoryId);
         }
     }
 
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        saveState(outState);
+        super.onSaveInstanceState(outState);
+    }
+
+    private void restoreState(Bundle savedInstanceState) {
+        List<Star> savedStars = savedInstanceState.getParcelableArrayList(STARS_KEY);
+        starsListPresenter.loadSavedStars(savedStars);
```

Comment:
just pass the savedInstanceState.getParcelableArrayList to the presenter and validate on the presenter